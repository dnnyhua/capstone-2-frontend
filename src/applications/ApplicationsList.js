import React from "react";
import ApplicationCard from "./ApplicationCard";
import './ApplicationsList.css'


const ApplicationsList = ({ applications, getApplications }) => {

    if (!applications) {
        return (
            <div>
                <h1>Loading...</h1>
            </div>
        )
    }

    console.log(applications)

    return (
        <div className="ApplicationsList">
            <h1>Applications</h1>
            {applications.map(application =>
                <ApplicationCard
                    key={`${application.jobId}${application.walkerId}`}
                    jobId={application.jobId}
                    walkerId={application.walkerId}
                    status={application.status}
                    firstName={application.firstName}
                    lastName={application.lastName}
                    rate={application.rate}
                    bio={application.bio}
                    profileImage={application.profileImage}
                    getApplications={getApplications}

                />
            )}

        </div>
    )
}

export default ApplicationsList

