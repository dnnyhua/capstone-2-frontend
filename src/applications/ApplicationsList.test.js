import React from "react";
import { render } from "@testing-library/react";
import ApplicationsList from "./ApplicationsList";
import { MemoryRouter } from "react-router";

it("matches snapshot with applications", function () {
    let applications = [
        {
            "id": 1,
            "jobId": 2,
            "walkerId": 1,
            "firstName": "Johnny",
            "lastName": "Bravo",
            "rate": 20,
            "bio": "huge dog lover",
            "status": "Hired",
            "profileImage": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSRNF8-gfvx__gaeVMxKBA2uxYZWju8XBV3_P0Rmsl_9lAcFVJCts7MZbNbE0bnARq1FFE&usqp=CAU"
        },
        {
            "id": 3,
            "jobId": 2,
            "walkerId": 2,
            "firstName": "Tom",
            "lastName": "Holland",
            "rate": 25,
            "bio": "I have experience walking dogs of all sizes",
            "status": "Job filled",
            "profileImage": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSBIYoCdF6ixsv0vzgejZMsBkpTYbh93v-67nbZk_hj2NBjVa5j3zrLMGr17T9OBMqWVzA&usqp=CAU"
        }
    ];
    const { asFragment } = render(
        <MemoryRouter>
            <ApplicationsList applications={applications} />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});


it("matches snapshot without applications", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <ApplicationsList />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

