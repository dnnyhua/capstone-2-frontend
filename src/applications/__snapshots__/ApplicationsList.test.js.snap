// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with applications 1`] = `
<DocumentFragment>
  <div
    class="ApplicationsList"
  >
    <h1>
      Applications
    </h1>
    <div
      class="ApplicationCard hired"
    >
      <div
        class="ApplicationCard-body"
      >
        <section>
          <img
            alt="walker"
            class="ApplicationCard-img"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSRNF8-gfvx__gaeVMxKBA2uxYZWju8XBV3_P0Rmsl_9lAcFVJCts7MZbNbE0bnARq1FFE&usqp=CAU"
          />
        </section>
        <section>
          <h5>
            Job Id: 2 
          </h5>
          <h5>
            Walker Id: 1
          </h5>
          <h5>
            Johnny Bravo
          </h5>
          <h5>
            Rate: $20 / 30 min
          </h5>
          <h5>
            About Me: huge dog lover
          </h5>
        </section>
        <section>
          <button
            class="btn btn-secondary ApplicationCard-hireBtn"
            disabled=""
          >
            Hire
          </button>
          <button
            class="btn btn-secondary ApplicationCard-rejectBtn"
            disabled=""
          >
            Reject
          </button>
        </section>
      </div>
      <div
        class="acknowledgmentModal "
      >
        <div
          class="modal-dialog"
        >
          <div
            class="modal-content "
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
              >
                Acknowledgement
              </h5>
            </div>
            <div
              class="modal-body"
            >
              <p>
                Once hired, you cannot select a different walker. Are you sure you want to hire this walker?
              </p>
            </div>
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Yes
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                No
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ApplicationCard "
    >
      <div
        class="ApplicationCard-body"
      >
        <section>
          <img
            alt="walker"
            class="ApplicationCard-img"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSBIYoCdF6ixsv0vzgejZMsBkpTYbh93v-67nbZk_hj2NBjVa5j3zrLMGr17T9OBMqWVzA&usqp=CAU"
          />
        </section>
        <section>
          <h5>
            Job Id: 2 
          </h5>
          <h5>
            Walker Id: 2
          </h5>
          <h5>
            Tom Holland
          </h5>
          <h5>
            Rate: $25 / 30 min
          </h5>
          <h5>
            About Me: I have experience walking dogs of all sizes
          </h5>
        </section>
        <section>
          <button
            class="btn btn-secondary ApplicationCard-hireBtn"
            disabled=""
          >
            Hire
          </button>
          <button
            class="btn btn-secondary ApplicationCard-rejectBtn"
            disabled=""
          >
            Reject
          </button>
        </section>
      </div>
      <div
        class="acknowledgmentModal "
      >
        <div
          class="modal-dialog"
        >
          <div
            class="modal-content "
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
              >
                Acknowledgement
              </h5>
            </div>
            <div
              class="modal-body"
            >
              <p>
                Once hired, you cannot select a different walker. Are you sure you want to hire this walker?
              </p>
            </div>
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Yes
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                No
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without applications 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Loading...
    </h1>
  </div>
</DocumentFragment>
`;
